aws-android-sdk-s3/src/main/java/com/amazonaws/mobileconnectors/s3/transferutility/TransferUtility.java

```diff
@@ -762,6 +843,32 @@ private int createMultipartUploadRecords(String bucket, String key, File file, O
         return dbUtil.bulkInsertTransferRecords(valuesArray);
     }
 
+    private File writeInputStreamToFile(InputStream inputStream) throws IOException {
+        if (inputStream == null) {
+            throw new IllegalArgumentException("Invalid inputStream: " + inputStream);
+        }
+
+        // Saves the data as a file in the temporary directory
+        File file = File.createTempFile("aws-s3", ".tmp");
+        if (!file.isFile()) {
+            throw new IOException("Error creating a temporary file.");
+        }
+        try (OutputStream outStream = new FileOutputStream(file)) {
+            byte[] buffer = new byte[1024];
+            int bytesRead;
+            // Keep reading until reaches the end of the stream
+            while ((bytesRead = inputStream.read(buffer)) != -1) {
+                outStream.write(buffer, 0, bytesRead);
+                outStream.flush();
+            }
+            inputStream.close();
+        } catch (IOException e) {
+            file.delete();
+            throw new IOException("Error writing the inputStream into a file.", e);
```

Comment:
Single letter variable names are hard to find with search tools. I recommend to avoid the `e`:
```suggestion
        } catch (IOException ioException) {
            file.delete();
            throw new IOException("Error writing the inputStream into a file.", ioException);
```

Other thought: you are catching an `IOException`, only to throw another `IOException`. Maybe you should just let the original exception bubble up, without catching it?

Reply:
How to do that? Do you mean just remove the catch block? Or just do `throw ioException;` in the catch block?